wave_lang/kernel/wave/codegen/handlers.py

```diff
@@ -197,17 +197,24 @@ def handle_allocate(emitter: WaveEmitter, node: fx.Node):
     except ValueError as e:
         raise ValidationError("Malformed arguments") from e
 
-    assert (
-        tail_padding == 0
-    ), "Tail padding must be handled by the minimize_shared_allocs pass"
-
     memref_shape = cast_py_literal(emitter, distributed_shape)
     element_type = IrType.parse(dtype.ir_type_asm())
     address_space = Attribute.parse("#gpu.address_space<workgroup>")
     memref_type = MemRefType.get(memref_shape, element_type, None, address_space)
 
     if parent is not None:
         parent = cast_py_value(emitter, parent).ir_value
+    elif tail_padding != 0:
```

Comment:
It's unfortunate there is no easy way we can have a common handling for tail padding, but since I have no better suggestions going to approve for now, we should in parallel keep thinking of better ideas :)